builder/builder-next/exporter/wrapper.go

```diff
@@ -0,0 +1,71 @@
+package exporter
+
+import (
+	"context"
+	"strings"
+
+	"github.com/docker/docker/builder/builder-next/exporter/overrides"
+	"github.com/moby/buildkit/exporter"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type ImageExportedByBuildkit interface {
+	ImageExportedByBuildkit(ctx context.Context, id string, desc ocispec.Descriptor) error
```

Comment:
Any benefits of defining `ImageExportedByBuildkit` as an interface and having the daemon implement it over a more explicit approach such as 
```
type BuildkitExportCallback func(...) error
```

and having the daemon pass in a function?

Reply:
In this case I don't think there's any benefit. Changed to your suggestion as it's a bit less mouthful.